### PR TITLE
Update example

### PR DESCRIPTION
**Pull request by mit-mit**
Fixes https://github.com/dart-lang/dart-docker/issues/4

Some tweaks to the example/examples:

* Remove the Functions example (I'd like if we kept this repo very foundational, and not pointing to a single Cloud product
* Update the Dockerfile steps as discussed in https://github.com/dart-lang/dart-docker/issues/4
* Add the sample app rather than call the generator (allows for browsing the sample code directly in github)
* Add a readme with instructions

### CODE REVIEW

File: example/pubspec.lock [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Generated by pub
```
Comment (kevmoo) [inline example/pubspec.lock:1]: shouldn't commit this, right?

Comment (mit-mit) [reply example/pubspec.lock:1]: Why not; this example is an app?

Comment (kevmoo) [reply example/pubspec.lock:1]: Because it'll have random deltas as we work on it and it adds no benefit to the end user

Comment (subfuzion) [reply example/pubspec.lock:1]: I think the example will very rarely change, so I don't think the deltas will be a big deal. Any example should reflect best practices for an app. However, for me the question is if we really want to maintain examples here at all (as opposed to a dedicated samples repo). I'm in favor of keeping this repo fairly pristine, showing the Dockerfile for this in the README that we'll publish to [official docs](https://github.com/docker-library/docs) under `dart`, but linking to a sample repo that we maintain elsewhere.

**Comment by mit-mit** [comment]
Abandoning; will work on adding similar content to `dart create`, and created a new PR for updating README: https://github.com/dart-lang/dart-docker/pull/10

**Pull request closed without merging**